modules/manage/pages/iceberg/query-iceberg-topics.adoc

```diff
@@ -0,0 +1,101 @@
+
+.`schema.avsc`
+[,avro]
+----
+{
+    "type" : "record",
+    "namespace" : "com.redpanda.examples.avro",
+    "name" : "ClickEvent",
+    "fields" : [
+       { "name": "user_id", "type" : "int" },
+       { "name": "event_type", "type" : "string" },
+       { "name": "ts", "type": "string" }
+    ]
+ }
+----
+
+You can register the schema under the `ClickEvent-value` subject:
+
+[,bash]
+----
+rpk registry schema create ClickEvent-value --schema path/to/schema.avsc --type avro
+----
+
+You can then produce to the `ClickEvent` topic using the following format:
+
+[,bash]
+----
+echo '"key1" {"user_id":2324,"event_type":"BUTTON_CLICK","ts":"2024-11-25T20:23:59.380Z"}' | rpk topic produce ClickEvent --format='%k %v\n' --schema-id=topic
+----
+
+The following SQL query returns values from columns in the `ClickEvent` table, with the table structure derived from the schema, and column names matching the schema fields. If you've integrated a catalog, query engines such as Spark SQL provide Iceberg integrations that allow easy discovery and access to existing Iceberg tables in object storage.
+
+[,sql]
+----
+SELECT user_id, 
+    event_type, 
+    ts 
+FROM <catalog-name>.ClickEvent;
+----
+
+[,bash,role=no-copy]
+----
++---------+--------------+--------------------------+
+| user_id | event_type   | ts                       |
++---------+--------------+--------------------------+
+| 2324    | BUTTON_CLICK | 2024-11-25T20:23:59.380Z |
++---------+--------------+--------------------------+
+----
+
+== Query topic in key-value mode
+
+In `key_value` mode, you do not associate the topic with a schema in the Schema Registry, which means using semi-structured data in Iceberg. 
+
+In this example, it is assumed you have created the `ClickEvent_key_value` topic, set `redpanda.iceberg.mode` to `key_value`, and are also connecting to a REST-based Iceberg catalog.
+
+You can produce to the `ClickEvent_key_value` topic using the following format:
+
+[,bash]
+----
+echo 'key1 {"user_id":2324,"event_type":"BUTTON_CLICK","ts":"2024-11-25T20:23:59.380Z"}' | rpk topic produce ClickEvent_key_value --format='%k %v\n'
+----
+
+This example queries the semi-structured data in the `ClickEvent_key_value` table, which also consists of another column `redpanda` containing the record key and other metadata:
+
+[,sql]
+----
+SELECT 
+    value
+FROM <catalog-name>.ClickEvent_key_value;
+----
+
+[,bash,role=no-copy]
+----
++------------------------------------------------------------------------------+
+| value                                                                        |
++------------------------------------------------------------------------------+
+| {"user_id":2324,"event_type":"BUTTON_CLICK","ts":"2024-11-25T20:23:59.380Z"} |
++------------------------------------------------------------------------------+
+----
+
```

Comment:
If we're looking for a place to mention snapshot expiry, it might be worth mentioning here, with an example of using a time travel query https://iceberg.apache.org/docs/latest/spark-queries/#sql

Snapshot expiry will trim down the window available for time travel in order to maintain a lower metadata size

Reply:
Updated